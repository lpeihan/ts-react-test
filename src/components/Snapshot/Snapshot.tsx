import React, { Component } from 'react'

interface SnapshotState {
  value: string
}

export default class Snapshot extends Component<{}, SnapshotState> {
  state = {
    value: "snapshot1"
  }

  render() {
    return (
      <p>{this.state.value}</p>
    )
  }
}