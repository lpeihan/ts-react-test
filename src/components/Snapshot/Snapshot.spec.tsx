import React from 'react';
import {render, cleanup} from '@testing-library/react';
import Snapshot from './Snapshot';

afterEach(cleanup);
 
describe("<Snapshot />", () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Snapshot />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
