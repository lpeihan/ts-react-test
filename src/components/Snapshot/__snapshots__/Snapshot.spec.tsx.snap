// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snapshot /> should take a snapshot 1`] = `
<DocumentFragment>
  <p>
    snapshot1
  </p>
</DocumentFragment>
`;
